### PR TITLE
Add inplace support for rename_axis

### LINKED ISSUE #15704: rename_axis inplace=True
#### Code Sample, a copy-pastable example if possible

```python
import pandas as pd
import numpy as np

df = pd.DataFrame({'a':np.random.random(10),'b':np.random.random(10)})
df
df.rename_axis('numbers', inplace=True)
df
```
#### Problem description

df.rename_axis with the inplace=True should return None and make the named axis change the index name in the original object.  However, up listing df again, no index name is present.

#### Expected Output
df.index.name should return 'numbers'
#### Output of ``pd.show_versions()``

<details>
INSTALLED VERSIONS
------------------
commit: None
python: 3.5.2.final.0
python-bits: 64
OS: Linux
OS-release: 4.4.0-66-generic
machine: x86_64
processor: x86_64
byteorder: little
LC_ALL: None
LANG: en_US.UTF-8

pandas: 0.18.1
nose: 1.3.7
pip: 9.0.1
setuptools: 27.2.0
Cython: 0.24.1
numpy: 1.11.1
scipy: 0.18.1
statsmodels: 0.6.1
xarray: None
IPython: 5.1.0
sphinx: 1.4.6
patsy: 0.4.1
dateutil: 2.5.3
pytz: 2016.6.1
blosc: None
bottleneck: 1.1.0
tables: 3.2.3.1
numexpr: 2.6.1
matplotlib: 1.5.3
openpyxl: 2.3.2
xlrd: 1.0.0
xlwt: 1.1.2
xlsxwriter: 0.9.3
lxml: 3.6.4
bs4: 4.5.1
html5lib: 0.999999999
httplib2: None
apiclient: None
sqlalchemy: 1.0.13
pymysql: None
psycopg2: None
jinja2: 2.8
boto: 2.42.0
pandas_datareader: 0.3.0.post

</details>

### PR DESCRIPTION
**Pull request by hendrikmakait**
- [x] closes #15704 
 - [x] tests added / passed
 - [x] passes ``git diff upstream/master --name-only -- '*.py' | flake8 --diff``

### CODE REVIEW

File: pandas/core/generic.py [hunk 1]
```diff
@@ -784,27 +784,29 @@ def rename_axis(self, mapper, axis=0, copy=True, inplace=False):
...
+        Alter the name or names of the axis.
 
         Parameters
         ----------
         name : str or list of str
             Name for the Index, or list of names for the MultiIndex
         axis : int or str
            0 or 'index' for the index; 1 or 'columns' for the columns
+        inplace : bool
+            whether to modify `self` directly or return a copy
```
Comment (jreback) [inline pandas/core/generic.py]: versionadded tag

File: pandas/core/generic.py [hunk 2]
```diff
@@ -831,9 +833,11 @@ def _set_axis_name(self, name, axis=0):
         axis = self._get_axis_number(axis)
         idx = self._get_axis(axis).set_names(name)
 
-        renamed = self.copy(deep=True)
+        inplace = validate_bool_kwarg(inplace, 'inplace')
+        renamed = self if inplace else self.copy(deep=True)
```
Comment (jreback) [inline pandas/core/generic.py]: deep=True is the default so can remove that

**Review by jreback — CHANGES_REQUESTED** [review]

File: pandas/core/generic.py (hunk 1 above)
Comment (HndrkMkt) [inline pandas/core/generic.py]: Which version should I specify?

Comment (HndrkMkt) [inline pandas/core/generic.py]: Specified 0.20.2

Comment (jreback) [inline pandas/core/generic.py]: move to 0.21.0

File: pandas/tests/frame/test_alter_axes.py [hunk 3]
```diff
@@ -418,6 +418,13 @@ def test_rename(self):
                               pd.Index(['bar', 'foo'], name='name'))
         assert renamed.index.name == renamer.index.name
 
+    def test_rename_axis(self):
+        renamed = self.frame.rename_axis('foo')
```
Comment (jreback) [inline pandas/tests/frame/test_alter_axes.py]: add the issue number as a comment

File: pandas/tests/frame/test_alter_axes.py [hunk 4]
```diff
@@ -418,6 +418,13 @@ def test_rename(self):
                               pd.Index(['bar', 'foo'], name='name'))
         assert renamed.index.name == renamer.index.name
 
+    def test_rename_axis(self):
```
Comment (jreback) [inline pandas/tests/frame/test_alter_axes.py]: test_rename_axis_inplace

File: pandas/tests/frame/test_alter_axes.py [hunk 5]
```diff
@@ -418,6 +418,13 @@ def test_rename(self):
...
+        renamed = self.frame.rename_axis('foo')
+        no_return = self.frame.rename_axis('foo', inplace=True)
+        assert no_return is None
+        renamed2 = self.frame
+        assert_frame_equal(renamed, renamed2)
```
Comment (jreback) [inline pandas/tests/frame/test_alter_axes.py]: make this read

``assert_frame_equal(result, expected)`` (rename things)

**Review by jreback — APPROVED** [review]

File: pandas/core/generic.py [hunk 6]
```diff
@@ -831,9 +835,11 @@ def _set_axis_name(self, name, axis=0):
         axis = self._get_axis_number(axis)
         idx = self._get_axis(axis).set_names(name)
 
-        renamed = self.copy(deep=True)
+        inplace = validate_bool_kwarg(inplace, 'inplace')
+        renamed = self if inplace else self.copy()
```
Comment (jorisvandenbossche) [inline pandas/core/generic.py]: Is there a reason you removed the `deep=True` ?

Comment (HndrkMkt) [inline pandas/core/generic.py]: It's the default and can thus be removed (as of @jreback's request that disappeared after my rebase).
https://github.com/pandas-dev/pandas/blob/master/pandas/core/generic.py#L3523

Comment (jorisvandenbossche) [inline pandas/core/generic.py]: OK, perfect! (I looked for a comment about it, but didn't find one :-))

**Review by jorisvandenbossche — APPROVED** [review]

**Review by jreback — CHANGES_REQUESTED** [review]
pls add a whatnews note in other enhancements for 0.21.0. ping on green.

File: pandas/tests/frame/test_alter_axes.py [hunk 7]
```diff
@@ -418,6 +418,14 @@ def test_rename(self):
                               pd.Index(['bar', 'foo'], name='name'))
         assert renamed.index.name == renamer.index.name
 
+    def test_rename_axis_inplace(self):
```
Comment (jreback) [inline pandas/tests/frame/test_alter_axes.py]: can you also add a similar test in series/test_alter_axes.py

and tests this with axis=1 as well.

File: doc/source/whatsnew/v0.21.0.txt [hunk 8]
```diff
@@ -33,6 +33,7 @@ Other Enhancements
 - The ``validate`` argument for :func:`merge` function now checks whether a merge is one-to-one, one-to-many, many-to-one, or many-to-many. If a merge is found to not be an example of specified merge type, an exception of type ``MergeError`` will be raised. For more, see :ref:`here <merging.validation>` (:issue:`16270`)
 - ``Series.to_dict()`` and ``DataFrame.to_dict()`` now support an ``into`` keyword which allows you to specify the ``collections.Mapping`` subclass that you would like returned.  The default is ``dict``, which is backwards compatible. (:issue:`16122`)
 - ``RangeIndex.append`` now returns a ``RangeIndex`` object when possible (:issue:`16212`)
+- ``Series.rename_axis()`` and ``DataFrame.rename_axis()`` with ``inplace=True`` now return None while renaming the axis inplace. (:issue:`15704`)
```
Comment (HndrkMkt) [inline doc/source/whatsnew/v0.21.0.txt]: @jreback - added

**Review by jreback — APPROVED** [review]

**Comment by jreback** [comment]
thanks @HndrkMkt

**Pull request merged**